comp/core/workloadmeta/collectors/internal/kubemetadata/kubemetadata.go

```diff
@@ -229,10 +231,35 @@ func (c *collector) parsePods(
 			}
 		}
 
-		var nsLabels map[string]string
-		nsLabels, err = c.getNamespaceLabels(pod.Metadata.Namespace)
-		if err != nil {
-			log.Debugf("Could not fetch namespace labels for pod %s/%s: %v", pod.Metadata.Namespace, pod.Metadata.Name, err)
+		var nsLabels, nsAnnotations map[string]string
+
+		if c.dcaClient.Version().Major >= 7 && c.dcaClient.Version().Minor >= 55 {
```

Comment:
I think it makes sense to put this logic in the client.
We could expose a function called `SupportsNamespaceMetadataCollection` or similar.
What do you think?

Reply:
I was following the same pattern that is already used [here](https://github.com/DataDog/datadog-agent/blob/92c5677509b1e0c2928dea93454b20f789463495/comp/core/workloadmeta/collectors/internal/kubemetadata/kubemetadata.go#L189).

But I agree it can be cleaner to expose a method as such in the client 👍